### PR TITLE
doc: protocol_serialization: Cleanup of documentation for protocol ser.

### PR DESCRIPTION
**Pull request by canisLupus1313**
Commit cleans protocol serialization docs.

### CODE REVIEW

File: doc/nrf/protocols/protocol_serialization/index.rst [hunk 1]
```diff
@@ -0,0 +1,280 @@
+.. _ug_protocols_serializations:
```
Comment (Damian-Nordic) [inline doc/nrf/protocols/protocol_serialization/index.rst:1]: This should be README.rst file in the application (current `server`).

File: doc/nrf/images/user_guide.rst [hunk 2]
```diff
@@ -111,7 +111,7 @@ The `uart1` peripheral is configured to use the following pins:
```
Comment (Damian-Nordic) [inline doc/nrf/images/user_guide.rst:111]: Is this file still needed after adding another RST that duplicates this?

**Review by Damian-Nordic — COMMENTED** [review]
General questions first. Also, the `ps_nrf_*.png` files seem to be in needlessly high resolution, so consider downscaling them to have less impact on the repo size :)

**Comment by canisLupus1313** [comment]
> General questions first. Also, the `ps_nrf_*.png` files seem to be in needlessly high resolution, so consider downscaling them to have less impact on the repo size :)

Yes I know those will be scaled to size.

File: doc/nrf/protocols/protocol_serialization/index.rst [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+
+#. Retrieve the operational dataset of the Thread network:
+
+    .. code-block:: console
+
+      uart:~$ ot dataset active -x
+      0e080000000000000000000300000b35060004001fffe00208dead00beef00cafe0708fddead00beef00000510f7893f15a55d8adeacad288c38bf32cc030a4f70656e546872656164010240120410d48d777a474f80e61aa5680de764bd6d0c0402a0f7f8
+      Done
+
+#. To send a UDP packet to a peer device and UDP port 5555, run the following command:
```
Comment (alxelax) [inline doc/nrf/protocols/protocol_serialization/index.rst:269]: Just note. With enabled debug snippet, I wasn't able to pass the last two steps since IP address is overwritten by debug information on screen. Customer has to keep in mind IP address, that is inconvenient :)

**Review by alxelax — COMMENTED** [review]
Please take a look at my comments. Also, it is not clear what to do with compliance issues. Seems pictures requires better compression.
